config/config.go

```diff
@@ -1022,6 +1022,7 @@ type RemoteWriteConfig struct {
 	Name                 string            `yaml:"name,omitempty"`
 	SendExemplars        bool              `yaml:"send_exemplars,omitempty"`
 	SendNativeHistograms bool              `yaml:"send_native_histograms,omitempty"`
+	SendWALMetadata      bool              `yaml:"send_metadata,omitempty"` // TODO(@tpaschalis) Adding an extra field to enable us to remove the `metadata_config` struct in the future.
```

Comment:
Let's start cleaning those TODOs. I would kill that comment and add small one:

Also we have a bit of inconsistency. Do we want `send_wal_metadata` or without WAL?  I assume `SendWALMetadata` and reviewed accordingly.

```suggestion
	// SendWALMetadata causes metadata from WAL to be propagated in Remote Write 2.0.
	SendWALMetadata      bool              `yaml:"send_wal_metadata,omitempty"
```
And instead add `// MetadataConfig is deprecated in favor of SendWALMetadata.` on top of MetadataConfig field.